### PR TITLE
docs: edit $any() section and add example in Template Syntax

### PR DESCRIPTION
**Pull request by kapunahelewong**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
Edits to copy and creation of a new example just for this section. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

This is the 11th (and last) in a series of PRs that together are a complete rewrite of Template Syntax. Once they are all merged, the next step is to break them into individual docs as in the NgModules section.

The with this PR and the other 9 that are yet to be merged, the Template Syntax rewrite is complete! 🎉The only remaining task is to then break them apart! 🌷

### CODE REVIEW

File: aio/content/examples/built-in-template-functions/e2e/src/app.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+'use strict'; // necessary for es6 output in node
+
+import { browser, element, by } from 'protractor';
+
+describe('Built-in-template-functions-example', function () {
```
Comment (brandonroberts) [inline aio/content/examples/built-in-template-functions/e2e/src/app.e2e-spec.ts]:
```suggestion
describe('Built-in Template Functions Example', function () {
```

File: aio/content/examples/built-in-template-functions/e2e/src/app.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    browser.get('');
+  });
+
+  it('should have title Built-in Template Functions', function () {
+    let title = element.all(by.css('h1')).get(0);
+    expect(title.getText()).toEqual('Built-in Template Functions');
+  });
+
+  it('should display $any( ) in h2', function () {
+    let jsonDate = element(by.css('h2'));
```
Comment (brandonroberts) [inline aio/content/examples/built-in-template-functions/e2e/src/app.e2e-spec.ts]: jsonDate?

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: aio/content/guide/template-syntax.md [hunk 3]
```diff
@@ -2009,29 +2009,29 @@ You'll need this template operator when you turn on strict null checks. It's opt
 
 <hr/>
 
+{@a built-in-template-functions}
+
+## Built-in template functions
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/template-syntax.md]: Added that it's during AoT and linked to it.

File: aio/content/examples/built-in-template-functions/e2e/src/app.e2e-spec.ts (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/examples/built-in-template-functions/e2e/src/app.e2e-spec.ts]: got it, thanks!

File: aio/content/examples/built-in-template-functions/e2e/src/app.e2e-spec.ts (hunk 2 above)
Comment (kapunahelewong) [reply aio/content/examples/built-in-template-functions/e2e/src/app.e2e-spec.ts]: Ha! That's unintended. Fixed to something that makes sense. Thank you!

**Review by brandonroberts — CHANGES_REQUESTED** [review]
Delete the `aio/content/examples/.DS_Store` file.

**Comment by kapunahelewong** [comment]
> Delete the `aio/content/examples/.DS_Store` file.

Gah! That thing snuck by me! Thanks for catching it. 😅

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Caretaker note: Using a global approval for a docs only change.

**Pull request closed without merging**